pkg/providers/aws.go

```diff
@@ -155,3 +180,106 @@ var _ = Describe("MetadataServiceOptions", framework.LabelCloudProviderSpecific,
+		var err error
+		client, err = framework.LoadClient()
+		Expect(err).ToNot(HaveOccurred(), "Failed to load client")
+
+		gatherer, err = framework.NewGatherer()
+		Expect(err).ToNot(HaveOccurred(), "Failed to load gatherer")
+
+		ctx = framework.GetContext()
+
+		platform, err := framework.GetPlatform(ctx, client)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get platform")
+		if platform != configv1.AWSPlatformType {
+			Skip(fmt.Sprintf("skipping AWS specific tests on %s", platform))
+		}
+		// Make sure to clean up the resources we created
+		DeferCleanup(func() {
+			Expect(framework.DeleteMachineSets(client, toDelete...)).To(Succeed())
+			toDelete = make([]*machinev1.MachineSet, 0, 3)
+
+			framework.WaitForMachineSetsDeleted(ctx, client, toDelete...)
+		})
+	})
+
+	AfterEach(func() {
+		specReport := CurrentSpecReport()
+		if specReport.Failed() {
+			Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed())
+		}
+	})
+
+	createMachineSet := func(capacityReservationId string) (*machinev1.MachineSet, error) {
+		var err error
+
+		By(fmt.Sprintf("Create machine with capacityReservationId %s", capacityReservationId))
+		machineSetParams := framework.BuildMachineSetParams(ctx, client, 1)
+		spec := machinev1.AWSMachineProviderConfig{}
+		Expect(json.Unmarshal(machineSetParams.ProviderSpec.Value.Raw, &spec)).To(Succeed())
+
+		spec.CapacityReservationID = capacityReservationId
+
+		machineSetParams.ProviderSpec.Value.Raw, err = json.Marshal(spec)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get MachineSet parameters")
+
+		mc, err := framework.CreateMachineSet(client, machineSetParams)
+		if err != nil {
+			return nil, err
+		}
+		toDelete = append(toDelete, mc)
+		framework.WaitForMachineSet(ctx, client, mc.GetName())
+
+		return mc, nil
+	}
+
+	// Machines required for test: 0
+	// No machines are created, because the machineSet is rejected.
+	It("should not allow to create machineset with incorrect capacityReservationId", func() {
+		_, err := createMachineSet("fooobaar")
+		Expect(err).To(HaveOccurred(), "Expected error, shouldn't be able to create machineSet with incorrect capacityReservationId")
+		Expect(err.Error()).Should(ContainSubstring("invalid value for capacityReservationId: \"fooobaar\", it must start with 'cr-' and be exactly 20 characters long with 17 hexadecimal characters"))
+	})
+
+	// Machines required for test: 1
+	It("machine should get Running with active capacityReservationId", func() {
+		By("Get instanceType and availabilityZone from the first worker MachineSet")
+		workers, err := framework.GetWorkerMachineSets(ctx, client)
+		Expect(err).ToNot(HaveOccurred())
+		worker0 := workers[0]
+		var awsProviderConfig machinev1.AWSMachineProviderConfig
+		err = json.Unmarshal(worker0.Spec.Template.Spec.ProviderSpec.Value.Raw, &awsProviderConfig)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Access AWS to create CapacityReservation")
+		oc, _ := framework.NewCLI()
+		getAwsCredentialFromCluster(oc)
```

Comment:
I would expect this function to return the credentials considering its name.
As an alternative approach we could then format the credentials and pass them the the `framework.InitAWSSession(...)` as a config parameter in `ec2.New(...)`. It would be a bit more descriptive than setting the environment variables within the function, and have the client load them up that way.
That said it is only an alternative, so not a blocking deal.